up42/auth.py

```diff
@@ -247,8 +249,8 @@ def _request(
         try:
             if self.retry:
                 retryer = Retrying(
-                    stop=stop_after_attempt(1),  # TODO: Find optimal retry solution
-                    wait=wait_fixed(0),
+                    stop=(stop_after_attempt(2) | stop_after_delay(3)),
```

Comment:
Isn't this redundant? If 2 attempts are allowed and we wait 0.5 seconds for each of them, then the retry will stop after 1 s if If understand the different parameters correctly.

Reply:
I also feel the same, I guess a single stop method should be sufficient in this.

Reply:
Thought about it as a fallback, we do 2 retries, but quit if they take longer than 3 seconds, e.g. if there is a server issue? But maybe the 2 retries should just play out, will remove the second method.